masonry/src/passes/layout.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
```

Comment:
```suggestion

//! The layout pass, which provides the size and position of each widget
//! before any translations applied in [`compose`](super::compose).
//! Most of the logic for this pass happens in [`Widget::layout`] implementations,
//!
//! In the future, this might also support scaling the child widgets.

```

Reply:
I think scaling the child widgets would happen in compose.

Reply:
Maybe. It depends on what purpose the scaling has

Because for text, layout can depend on the absolute logical size of the text, due to the `opsz` (optical size) variable font axis. So it depends on whether the scaling is logically scaling the logic size

Reply:
Added, except for the part about scaling children.